src/nix/develop.cc

```diff
@@ -142,6 +147,18 @@ struct BuildEnvironment
 
     void toBash(std::ostream & out, const std::set<std::string> & ignoreVars) const
     {
+        if (minimalBashVersion.has_value()) {
+            out << stripIndentation(fmt(R"__NIX_STR(
+                            if [[ -n "${BASH_VERSINFO-}" && "${BASH_VERSINFO-}" -lt %d ]]; then
+                                echo "Detected Bash version that isn't supported by this derivation (${BASH_VERSION})"
+                                echo "Please install Bash %d or greater to continue."
+                                exit 1
```

Comment:
Maybe this should be a warning rather than a fatal error?

Reply:
Hmm, I'm not sure. The issue I was facing was that sourcing the script generated by `print-dev-env` with an old version of bash resulted in incomplete changes to the environment. This can be very disorienting (and annoying) for the user, as that can remove nix itself from PATH.

In the spirit of repeatability, I would say aborting fully if we can't guarantee the script is valid is the safest option.

Reply:
Making it a warning would be very confusing since it would very likely be immediately followed by some (apparently) nonsensical bash errors (and leave the shell in a weird state). Failing early seems to be the reasonable solution.